### PR TITLE
refactor(lib): add miranum-vscode lib

### PR DESCRIPTION
**Pull request by peterhnm**
**Issue**
We have reusable components in our VS Code Extensions that should be provided by a lib.

**Description**
* Crate a miranum-vscode lib with Nx
* Move all reusable components to that lib

**Checklist**

* Code is easy to understand
* No obvious errors or bugs
* CI/CD Pipelines did run successfully
* Tests were implemented
* Documentation was created

### CODE REVIEW

**Comment by peterhnm** [comment]
@lmoesle I created a lib for our VS Code Plugins and moved some parts of the *miranum-modeler* there. I then ran into a problem when using the library in a webview. I described the whole thing [here](https://github.com/FlowSquad/miranum-ide/blob/refactor/add_miranum-vscode_lib/libs/miranum-vscode/README.md). 
First, is it understandable what I wrote there? 
And second, what do you think about the workaround with the dummy module? 
As an alternative, I could create a second lib (e.g. miranum-vscode-webview) that only contains components that don't need an import of the VS Code API. In my opinion this is a overkill and leads to the problem that there might be confusion which lib to use.

File: libs/miranum-vscode/src/lib/miranum-vscode.ts
Comment (lmoesle) [inline libs/miranum-vscode/src/lib/miranum-vscode.ts:1]: The file just contains a Logger. We should rename it to `Logger.ts`

**Review by lmoesle — APPROVED** [review]

**Comment by lmoesle** [comment]
> @lmoesle I created a lib for our VS Code Plugins and moved some parts of the _miranum-modeler_ there. I then ran into a problem when using the library in a webview. I described the whole thing [here](https://github.com/FlowSquad/miranum-ide/blob/refactor/add_miranum-vscode_lib/libs/miranum-vscode/README.md). First, is it understandable what I wrote there? And second, what do you think about the workaround with the dummy module? As an alternative, I could create a second lib (e.g. miranum-vscode-webview) that only contains components that don't need an import of the VS Code API. In my opinion this is a overkill and leads to the problem that there might be confusion which lib to use.

Yes, I get it and the workaround is well explained.

I am not the biggest fan of the dummy module approach.  I would take the alternative and create multiple libs.
Nx supports different types of libs. There are buildable libs that are built to a standalone lib that may gets published to e.g. npmjs and the "default" libs which are just added to the applications bundle. The "default" libs are just helpers to structure the code.
I wouldn't publish the miranum-vscode lib and use it just internally in this repository. I don't see why it's confusing to have multiple libs (as long as their meaningful named). Most of the developers using them are familiar with our code base.

**Comment by peterhnm** [comment]
@lmoesle As you suggested I did create multiple libaries to solve the build problem with vite.

File: libs/vscode/miranum-vscode/src/lib/fileSystem/Reader.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import * as vscode from "vscode";
-import { Uri } from "vscode";
-import { FolderContent, WorkspaceFolder } from "../types";
-import { Logger } from "./Logger";
+import { Logger } from "../Logger";
+import { FolderContent } from "@miranum-ide/vscode/miranum-vscode-webview";
```
Comment (peterhnm) [inline libs/vscode/miranum-vscode/src/lib/fileSystem/Reader.ts:3]: @lmoesle I don't realy like this dependency. But I do need the data structure in the webview to extract it correctly. Maybe you have a better solution.

**Pull request merged**